### DISCUSSION TITLE
How to get version during runtime

### DISCUSSION THREAD

**Discussion by itavero** [adamralph/minver#495] [Q&A] [answered]
I'm building a console application, using [CliFx](https://github.com/Tyrrrz/CliFx) which uses  `Assembly.GetEntryAssembly` to [determine the version](https://github.com/Tyrrrz/CliFx/blob/94f63631db714edbc48119d4124968d5cda70e1b/CliFx/CliApplicationBuilder.cs#L208) of the application.

For some strange reason I can see the right values being filled in if I look into the file properties in Explorer, but the Assembly version does not show the correct value.

The last tag is `v0.0.3` (and I have `MinVerTagPrefix` set to `v` in the csproj file. MinVer picks this up without any problem:
```
0>MinVer: Using { Commit: 09ef89a, Tag: 'v0.0.3', Version: 0.0.3, Height: 0 }.
0>MinVer: Calculated version 0.0.3.
```

However the version being printed by CliFx (and also if I manually access the "entry assembly" from the immediate window during debugging), stays at `0.0.0.0`.
I've even added `<AssemblyVersion>$(MinVerVersion)</AssemblyVersion>` to the `PropertyGroup` in the csproj file, because I noticed in the README that is states that by default only the major version is reflected in here.

Does anyone have any clue what I'm missing here?

**Comment by itavero** [comment, marked as answer]
Okay.. my bad. I tried to set the `AssemblyVersion` in the wrong way.
After adding an additional target as was mentioned in the README (which I totally missed somehow), the `AssemblyVersion` is updated correctly.

```xml
    <Target Name="UpdateAssemblyVersion" AfterTargets="MinVer">
        <PropertyGroup>
            <AssemblyVersion>$(MinVerVersion)</AssemblyVersion>
        </PropertyGroup>
    </Target>
```

**Reply by jbennink** [reply]
In my case this did not work. Th generated assemblyinfo.cs tried to set the AssemblyVersion to a pre version. I guess this only works on RTM x.y.z.0 versions:
`
[assembly: System.Reflection.AssemblyFileVersionAttribute("5.0.1.0")]
[assembly: System.Reflection.AssemblyInformationalVersionAttribute("5.0.1-pre.0.47")]
[assembly: System.Reflection.AssemblyVersionAttribute("5.0.1-pre.0.47")]
`

**Reply by jbennink** [reply]
This howerver did work to set the AssemblyVersion
```
<Target Name="UpdateAssemblyVersion" AfterTargets="MinVer">
  <PropertyGroup>
	  <AssemblyVersion>$(MinVerMajor).$(MinVerMinor).$(MinVerPatch).0</AssemblyVersion>
  </PropertyGroup>
</Target>
```

**Reply by adamralph** [reply]
Yes, this is very similar to one of the examples in the [FAQ](https://github.com/adamralph/minver#can-i-use-the-version-calculated-by-minver-for-other-purposes).

**Reply by itavero** [reply]
Unfortunately for me this also broke and I had to change the AssemblyVersion as shown above by @jbennink.
I wonder, is there a way to get a string representation that I can use in my code that does have the full MinVerVersion (so including the pre-release info and such)?
It would be nice to at least be able to show the full version in the interface of the application.

**Reply by adamralph** [reply]
@itavero unless you're doing something special, the full version should flow through to [AssemblyInformationalVersionAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assemblyinformationalversionattribute?view=net-7.0).
